### PR TITLE
AG-1346 GCT bug fix for zero p-values showing wrong circle size

### PR DESCRIPTION
**Pull request by sagely1**

### CODE REVIEW

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts [hunk 1]
```diff
@@ -912,15 +912,20 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
     const tissue = gene.tissues.find((t) => t.name === tissueName);
     let size = 0;
     let color = '#F0F0F0';
+    const CIRCLE_MAX_SIZE = 50;
 
     if (tissue?.adj_p_val) {
       size = this.getCircleSize(tissue.adj_p_val);
     }
 
+    if (size === 0) {
```
Comment (sagely1) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts]: if circle size is zero, it should have the max size of circle as the p-value is inversely proportional to the size of the circle.

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts [hunk 2]
```diff
@@ -912,15 +912,20 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
...
+      size = CIRCLE_MAX_SIZE;
+    }
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts]: What do you think about handling this case in `getCircleSize` instead, so all of the sizing is determined in the same function?

Comment (sagely1) [reply src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts]: That's a good idea.  I refactored this method so there would be more clarity as there is too much logic spread out here for size and style.

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts [hunk 3]
```diff
@@ -898,36 +901,35 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
...
+      return 0;
+
+    // if significance cutoff radio button selected and 
+    // p-Value > significance threshhold, don't show
     if (this.significanceThresholdActive && pval > this.significanceThreshold) {
       return 0;
     }
 
     const pValue = 1 - (this.nRoot(pval, 3) || 0);
-    const size = Math.round(100 * pValue * 0.44);
```
Comment (sagely1) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts:907]: 100 * 0.44 = 44 (What was previously the max size)
Since the legend is 50px for the largest circle, let's just multiply the pValue by the MAX_SIZE for simplicity.

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts [hunk 4]
```diff
@@ -898,36 +901,35 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
...
-    return size < 6 ? 6 : size;
+    const size = Math.round(pValue * MAX_SIZE);
+
+    // ensure the smallest circles have a min size to be easily hoverable/clickable
+    return size < MIN_SIZE ? MIN_SIZE : size;
   }
 
   getCircleStyle(tissueName: string, gene: GCTGene) {
     const tissue = gene.tissues.find((t) => t.name === tissueName);
-    let size = 0;
```
Comment (sagely1) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts:913]: Methods are ideally small and clear so I refactored this for clarity

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts [hunk 5]
```diff
@@ -898,36 +901,35 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
...
-    if (tissue?.logfc) {
-      color = this.getCircleColor(tissue.logfc);
-    }
-      
+    const size = this.getCircleSize(tissue?.adj_p_val);
+    const color = this.getCircleColor(tissue?.logfc);
+    
     return {
-      display: size ? 'block' : 'none',
+      display: size > 0 ? 'block' : 'none',
```
Comment (sagely1) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts:932]: Now that size will always be defined as a number, show for values larger than zero.

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts [hunk 6]
```diff
@@ -1019,10 +1026,10 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
...
           ...[
             tissueName,
-            tissue ? tissue.logfc : '',
-            tissue ? tissue.ci_r : '',
-            tissue ? tissue.ci_l : '',
-            tissue ? tissue.adj_p_val : '',
+            tissue?.logfc || '',
+            tissue?.ci_r || '',
+            tissue?.ci_l || '',
+            tissue?.adj_p_val || '',
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts]: Was this change intentional? It looks like this may reintroduce the bug fixed in [AG-1347](https://github.com/Sage-Bionetworks/Agora/pull/1283)

**Review by hallieswan — COMMENTED** [review]
Thanks for the cleanup! The refactored methods are a lot clearer. Just one small question!

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts (hunk 6 above)
Comment (sagely1) [reply src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts]: Good catch! Not sure why this change didn't merge from AG-1347 but I committed it.

**Review by hallieswan — APPROVED** [review]
